src/targets/common.ini

```diff
@@ -40,6 +40,7 @@ build_flags =
 	-D PLATFORM_ESP32=1
 	-D CONFIG_TCPIP_LWIP=1
 	-D BEARSSL_SSL_BASIC
+	-I ${PROJECTSRC_DIR}/hal
```

Comment:
Not needed

Reply:
Do we wanna remove these? I added these for future usage...

Reply:
Any sub-folders you add to the lib folder is automatically added to the include path.

Reply:
I tried to add that hal file into libs but it wont compile it without any header... I can remove these include paths, no problem :)

Reply:
I just tested moving src/hal to lib/HAL and removing the `-I` but leaving the `HAL` library include and it compiles and links it in.

Reply:
Yeah, later I realize that stm common `lib_deps` is not include into all envs...